controllers/app.controllers.js

```diff
@@ -1,55 +1,128 @@
-const { getAllTopics, getAllUsernames, getArticleFromDbById, getAllArticles, getCommentsForArticle } = require('../models/app.models');
-
+const {
+  fetchAllTopics,
+  fetchAllUsernames,
+  fetchArticleFromDbById,
+  fetchAllArticles,
+  fetchCommentsForArticle,
+  sendCommentByArticleId,
+  removeCommentById,
+  checkUserExistsByName,
+  updateVotes,
+} = require('../models/app.models');
 
 exports.getTopics = async (req, res, next) => {
-    try {
-        const topics = await getAllTopics();
-        res.status(200).send({topics});
-    } catch(err) {
-        next(err);
-    }
-}
+  try {
+    const topics = await fetchAllTopics();
+    res.status(200).send({ topics });
+  } catch (err) {
+    next(err);
+  }
+};
 
 exports.getUsernames = async (req, res, next) => {
-    try {
-        const usernames = await getAllUsernames();
-        res.status(200).send({usernames});
-    } catch(err) {
-        next(err);
-    }
-}
-
+  try {
+    const usernames = await fetchAllUsernames();
+    res.status(200).send({ usernames });
+  } catch (err) {
+    next(err);
+  }
+};
 
 exports.getArticles = async (req, res, next) => {
     try {
-        const articles = await getAllArticles();
-        res.status(200).send({articles});
-    } catch {
-        next(err);
-    }
-}
+        
+    const { sort_by } = req.query;
+    const { order } = req.query;
+    const { topic } = req.query;
+    
+      if (order) {
+        const validOrders = ['ASC', 'ASCENDING', 'DESC', 'DESCENDING'];
+        if(!validOrders.includes(order) )
+          return next({status: 400, msg: 'Bad Request'})
+        }
+      
 
+    const articles = await fetchAllArticles(sort_by, order, topic);
+    res.status(200).send({ articles });
+  } catch(err) {
+    next(err);
+  }
+};
 
 exports.getArticleById = async (req, res, next) => {
-    const { article_id } = req.params;
-    try {
-        const article = await getArticleFromDbById(article_id);
-        res.status(200).send({article});
-    } catch(err) {
-        next(err);
-    }
-}
+  const { article_id } = req.params;
+  try {
+    const article = await fetchArticleFromDbById(article_id);
+    res.status(200).send({ article });
+  } catch (err) {
+    next(err);
+  }
+};
 
 exports.getCommentsForArticleById = async (req, res, next) => {
-    const { article_id } = req.params;
-    try {
-        const article =  await getArticleFromDbById(article_id);
-        if(article) {
-            const comments = await getCommentsForArticle(article_id);
-                res.status(200).send({comments}); 
-        }
-    } catch(err) {
-        next(err);
+  const { article_id } = req.params;
+  try {
+    const article = await fetchArticleFromDbById(article_id);
+    if (article) {
```

Comment:
What's the purpose of this if statement? 
If it doesn't happen your server will not send a response and hang which is a situation you never want to be in. 
If your model is going to reject on a missing article then you can just 
`await fetchArticleFromDbById(article_id)`
then carry on. No need for the article variable at all.